test/InitStateTest.t.sol

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "./InitState.t.sol";
+
+contract InitStateTest is InitState {
+
+    function testMint(uint256 x) public {
+        uint256 xMin = swap.x_0() / swap.y_0();
+        x = bound(x, xMin, 1e26);
+        uint256 y = x * swap.y_0() / swap.x_0();
+        uint256 xBal = iDAI.balanceOf(amuro);
+        uint256 yBal = iWETH.balanceOf(amuro);
+        uint256 zBal = swap.balanceOf(amuro);
+        uint256 z = x * swap.totalSupply() / swap.x_0();
+
+        vm.startPrank(amuro);
+        iDAI.approve(address(swap), 2**256-1);
+        iWETH.approve(address(swap), 2**256-1);
+        swap.addLiquidity(x);
+        vm.stopPrank();
+        assertEq(iDAI.balanceOf(amuro), xBal - x);
+        assertEq(iWETH.balanceOf(amuro), yBal - y);
+        assertEq(swap.balanceOf(amuro), zBal + z);
+    }
+
+    function testMintEmit(uint256 x) public {
+        uint256 xMin = swap.x_0() / swap.y_0();
+        x = bound(x, xMin, 1e26);
+        uint256 y = x * swap.y_0() / swap.x_0();
+        uint256 z = x * swap.totalSupply() / swap.x_0();
+
+        vm.startPrank(gharma);
+        iDAI.approve(address(swap), 2**256-1);
+        iWETH.approve(address(swap), 2**256-1);
+        vm.expectEmit(true, false, false, true);
+        emit LiquidityProvided(gharma, x, y, z);
+        swap.addLiquidity(x);
```

Comment:
You also need to test that all return values are what they should be. If you have a bug there and you don't test, you might not find out until someone tries to integrate with you.